### ISSUE TITLE
[Bug] Can't escaped a period ('.') in an anyOf call

### ISSUE THREAD

**Issue by PaulJPhilp** [callstack/ts-regex-builder#75]
**Describe the bug**
Due to the missing constructor (issue #74) I tried to use anyOf('.') in the following:

```
const period = anyOf('.')
const hostname = [host, optional(repeat([period, host], { min: 1, max: 255 }))];
```

The result was an unescaped period ('.') (match any character) in the regex.  This caused a number of weird issues
which kept me company all weekend.

I then tried an escaped period in the anyOf:

```
const period = anyOf('\.')
const hostname = [host, optional(repeat([period, host], { min: 1, max: 255 }))];
```

There was no change, the result was an unescaped period.

**To Reproduce**
```
import { buildRegExp, anyOf } from '..';

const period = anyOf('.');
const regex = buildRegExp(period);
console.log(regex.source); 
console.log(regex);

const escapedPeriod = anyOf('\.');
const escapedRegex = buildRegExp(escapedPeriod);
console.log(escapedRegex.source);
console.log(escapedRegex);

test('`plain period`', () => {
  expect(regex.source).toBe('.');
  expect(regex).toEqualRegex(RegExp(/./));
  expect(regex).toMatchString('.');
  expect(regex).not.toMatchString('a');
});

test('`escapedRegex`', () => {
  expect(escapedRegex.source).toBe('.');
  expect(escapedRegex).toEqualRegex(RegExp(/\./));
  expect(escapedRegex).toMatchString('.');
  expect(escapedRegex).toMatchString('a');
});
```
**Expected behavior**
I expected to match the period in hostnames (i.e.  www.google.com).

**Screenshots**

**Package version**
`ts-regex-builder: ???`

**Comment by mdjastrzebski** [comment]
Good catch, TS regex optimizes single characters classes, which causes `.` to be come `any`. Simple mitigation would be to avoid using `anyOf(".")`, and just use `.`, there is no need to add `anyOf` for single character.

Eitherway, I will issue a bugfix for that.

**Comment by PaulJPhilp** [comment]
Agreed, I'll use "." from now on.

This was a nightmare bug to track down.  A fix is needed.

**Comment by mdjastrzebski** [comment]
Fixed by #76

**Comment by mdjastrzebski** [comment]
Resolved in [v1.4.0](https://github.com/callstack/ts-regex-builder/releases/tag/v1.4.0) 🎉
